File: android/sdk/src/main/java/org/elastos/sdk/elephantwallet/contact/internal/AcquireArgs.java
package org.elastos.sdk.elephantwallet.contact.internal;

class AcquireArgs {
    public enum Type {
        PublicKey(201),
        EncryptData(202),
        DecryptData(203),
        DidPropAppId(204),
        DidAgentAuthHeader(205),
        SignData(206);

        public static Type valueOf(int id) {
            Type[] values = Type.values();
            for(int idx = 0; idx < values.length; idx++) {
                if(values[idx].id == id) {
                    return values[idx];
                }
            }
            return null;
        }

        private Type(int id){
            this.id = id;
        }
        private int id;
    }

    public AcquireArgs(int type, String pubKey, byte[] data, String extra) {
        this.type = Type.valueOf(type);
        this.publicKey = pubKey;
        this.data = data;
        this.extra = extra;
    }

    @Override
    public String toString() {
        return "AcquireArgs"
             + "[type=" + type
             + ",publicKey=" + publicKey
             + ",data=" + data
             + ",extra=" + extra
             +"]";
    }

    public Type type;
    public String publicKey;
    public byte[] data;
    public String extra;
}

File: ios/sdk/swift/internal/AcquireArgs.swift

extension ContactInternal {

public class AcquireArgs {
  public enum Kind: Int {
    case PublicKey = 201
    case EncryptData = 202
    case DecryptData = 203
    case DidPropAppId = 204
    case DidAgentAuthHeader = 205
    case SignData = 206
  }
  
  public init(type: Int, pubKey: String?, data: Data?, extra: String?) {
    self.type = Kind(rawValue: type)!
    self.publicKey = pubKey
    self.data = data
    self.extra = extra
  }

  public func toString() -> String {
    return  "AcquireArgs"
          + "[type=\(type)"
          + ",publicKey=\(String(describing: publicKey))"
          + ",data=\(String(describing: data))"
          + ",extra=\(String(describing: extra))"
          + "]"
  }

  public let type: Kind
  public let publicKey: String?
  public let data: Data?
  public let extra: String?
}
  
}
